src/vs/editor/contrib/codeAction/browser/codeActionMenu.ts

```diff
@@ -139,18 +156,49 @@ class CodeMenuRenderer implements IListRenderer<ICodeActionMenuItem, ICodeAction
 			data.root.classList.add('group-header');
 		} else {
 			const text = element.action.label;
-			data.text.textContent = text;
 			element.isEnabled = element.action.enabled;
 
 			if (element.action instanceof CodeActionAction) {
+				const openedFromString = (element.params?.options.fromLightbulb) ? CodeActionTriggerSource.Lightbulb : element.params?.trigger.triggerAction;
+
 
 				// Check documentation type
 				element.isDocumentation = element.action.action.kind === CodeActionMenu.documentationID;
 
 				if (element.isDocumentation) {
-					data.text.textContent = text;
+					element.isEnabled = false;
 					data.root.classList.add('documentation');
+
+					const container = data.root;
+
+					const actionbarContainer = dom.append(container, dom.$('.codeActionWidget-action-bar'));
+
+					const reRenderAction = showDisabled ?
+						<IAction>{
+							id: 'hideMoreCodeActions',
+							label: 'Hide More',
+							enabled: true,
+							run: () => CodeActionMenu.toggleDisabledOptions(element.params)
```

Comment:
Can `toggleDisabledOptions()` take a parameter for enabled/disabled? With the current approach I'm worried the state may get out of sync

Reply:
I need 2 `<IAction>`  variables for sure just to deal with the localization of the variables, but the `enabled` here in `IAction` should always be true.